kayvee.go

```diff
@@ -33,3 +44,52 @@ func FormatLog(source string, level LogLevel, title string, data map[string]inte
 	data["title"] = title
 	return Format(data)
 }
+
+// SentryClient interface allows mocking for testing
+type sentryClient interface {
+	Capture(packet *raven.Packet, captureTags map[string]string) (eventID string, ch chan error)
+}
+
+// Logger provides an wrapper methods to do logging using kayvee.
+type Logger struct {
+	source       string
+	logger       *log.Logger
+	sentryClient sentryClient
+}
+
+// NewLogger returns a new *kayvee.Logger.
+// source is the value assigned for all logs generated by the logger
+// log.Logger is the underlying logger used. If nil, uses log.New(os.Stderr, "", log.Ldate|log.Ltime|log.Lshortfile)
+// sentryClient is used to optionally route errors to sentry.
+func NewLogger(source string, logger *log.Logger, sentryClient *raven.Client) *Logger {
+	if logger == nil {
+		logger = log.New(os.Stderr, "", defaultFlags)
+	}
+	return &Logger{source: source, logger: logger, sentryClient: sentryClient}
+}
+
+// LogInfo writes a log with level kayvee.Info
+func (l *Logger) LogInfo(title string, data map[string]interface{}) {
+	l.logger.Output(2, FormatLog(l.source, Info, title, data))
+}
+
+// LogWarning writes a log with level kayvee.Warning
+func (l *Logger) LogWarning(title string, data map[string]interface{}) {
+	l.logger.Output(2, FormatLog(l.source, Warning, title, data))
+}
+
+// LogError writes a log with level kayvee.Error
+// If the logger was initialized with a sentryClient and error is not nil, captures the error for
+// sentry and assigns the event ID to the `sentry_event_id` key in the data.
+func (l *Logger) LogError(title string, data map[string]interface{}, err error) {
+	sentryID := ""
+	if l.sentryClient != nil && err != nil {
+		packet := raven.NewPacket(err.Error(), raven.NewException(err, raven.NewStacktrace(1, 3, nil)))
+		sentryID, _ = l.sentryClient.Capture(packet, map[string]string{})
+	}
+	if sentryID != "" {
+		data["sentry_event_id"] = sentryID
+	}
```

Comment:
I moved the logic into a helper for code clarity. That said, I didn't see the immediate benefit of in tests because I would end up going through all the test cases anyways.